streamtools/consumers.py

```diff
@@ -74,6 +78,48 @@ def __call__(self, queue_arg="", override=False, *args, **kwargs):
 
         return self._decorator(set_queue_label, *args, **kwargs)
 
+    def parse_decorated_args(self, func, args_list, kwargs_list):
+        args_error_msg = "Please use only one argument (and optional 'self' arg in classes)" + \
+                         " for the message to be passed."
+
+        if kwargs_list:
```

Comment:
seems like these 2 ifs basically raise errors if you've called it with any args beyond a `func`?

Reply:
It enforces strict control of what you can pass into the decorated function's params. Pretty much this decorator is designed to decorate a function that only accepts a single message param (whether it be an http request or a consumed message off a queue.

I figured this would be easier given the wide set of places we're aiming to use this in. It also helps with the parsing of the `self` param inside class methods in the logic defined lower down here.